functions/checkAndImportRast.R

```diff
@@ -0,0 +1,20 @@
+# Define the path for data source
+checkAndImportRast <- function(parameter, regionGeometry, dataPath, fileType = "tiff") {
+  # Initialize the raster found flag
+  raster <- NULL
+  
+  # List all files in the directory that match the parameter
+  filePattern <- paste0(parameter, sprintf( "_.*\\.%s$", fileType))
+  fileList <- list.files(path = dataPath, pattern = filePattern, full.names = TRUE)
+  
+  # Check each file
+  for (file in fileList) {
+    rastObj <- terra::rast(file)
+    if (isSubset(regionGeometry, rastObj)) {
+      raster <- rastObj
+      cat(sprintf("Raster for '%s' encompassing region found and imported successfully!\n", parameter))
```

Comment:
Only problem now is here: when looking for distance_water this line returns "Raster for 'land_cover_corine' encompassing region found and imported successfully!". If this is a quick fix then go ahead and change, but if it would take a while to redo then let's just leave it as is.

Reply:
Also a general comemnt: try to use message(), warnings(), error() rather than cat(). The practical issue is that with cat() you can't suppress warnings etc.

Of course, this doesn't have to all be fixed now, but just something to keep in mind.

Reply:
done